### PR TITLE
[GEP-24] Fix: Configure DiscoveryServerDown alert only when discovery server is enabled

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area ipcei
/label ipcei/oidc 
/kind bug

**What this PR does / why we need it**:
[GEP-24] Fix: Configure DiscoveryServerDown alert only when discovery server is enabled.
#9746 unconditionally configured the `DiscoveryServerDown` alert which fires for gardens that does not have the gardener discovery server enabled. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

/cherry-pick release-v1.96

cc @dimityrmirchev 

**Release note**:

```bugfix operator
Fix a bug that unconditionally enables the `DiscoveryServerDown` alert even when the Gardener Discovery Server is not enabled. 
```

### CODE REVIEW

File: pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go [hunk 1]
```diff
@@ -68,7 +66,24 @@ func init() {
 }
 
 // CentralPrometheusRules returns the central PrometheusRule resources for the garden prometheus.
-func CentralPrometheusRules() []*monitoringv1.PrometheusRule {
+func CentralPrometheusRules(isGardenerDiscoveryServerEnabled bool) []*monitoringv1.PrometheusRule {
+	gardenTmpl, err := template.New("garden").Delims("<<", ">>").Parse(string(gardenYAML))
```
Comment (rfranzke) [inline pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go]: 😵 Let's avoid going down this "templating path" again if we can avoid it. Feel free to move this rule out of `garden.yaml` into a dedicated file (or just directly translate into a Golang `monitoringv1.PrometheusRule` object that can be conditionally added to the returned list). I just put this file there to easily adopt the existing rules, but this is in no way "state of the art" or something that should be extended if not trivial.

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go (hunk 1 above)
Comment (vpnachev) [reply pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go]: Done with a7f6d846

File: skaffold.yaml [hunk 2]
```diff
@@ -850,12 +850,7 @@ deploy:
               # TODO(timebertt): drop this after v1.96 has been released
               - bash
               - -ec
-              - |
-                until kubectl get controllerdeployments.v1.core.gardener.cloud >/dev/null; do 
-                  echo "Waiting for controllerdeployments.v1.core.gardener.cloud to be available..."
-                  sleep 5
-                done
-                echo "controllerdeployments.v1.core.gardener.cloud available"
+              - "until kubectl get controllerdeployments.v1.core.gardener.cloud >/dev/null; do \n  echo \"Waiting for controllerdeployments.v1.core.gardener.cloud to be available...\"\n  sleep 5\ndone\necho \"controllerdeployments.v1.core.gardener.cloud available\"\n"
```
Comment (rfranzke) [inline skaffold.yaml]: Why did you change this? 👀

File: pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go [hunk 3]
```diff
@@ -68,14 +65,176 @@ func init() {
...
+		return map[string]string{
+			"severity": severity,
+			"topology": "garden",
+		}
+	}
+
+	rules := []monitoringv1.Rule{
+		{
+			Alert:  "DashboardDown",
+			Expr:   intstr.FromString("probe_success{job = \"blackbox-dashboard\",purpose = \"availability\"} == 0"),
```
Comment (rfranzke) [inline pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go:91]: Nit: Please use `` instead of "" so that we don't have to escape the quotes in the query (not only here, but also in other `Expr`s and `description`s).

**Review by rfranzke — COMMENTED** [review]
Thanks @vpnachev, I actually wasn't asking you to "translate" the whole `garden.yaml` (it would have been enough to just move out the rule for GDS and put it into a dedicated `PrometheusRule`), but even better now. Thanks for the efforts 🚀

File: skaffold.yaml (hunk 2 above)
Comment (vpnachev) [reply skaffold.yaml]: I had to run https://github.com/gardener/gardener/blob/master/hack/update-skaffold-deps.sh and it changed the file.

File: pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go (hunk 3 above)
Comment (vpnachev) [reply pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go:91]: I wanted to do it like this, however the description has expressions that are already using the back quote which unfortunately cannot be escaped, it would require some mixed usage of back quote and double quotes like

```go
foo := `hello world
value = ` + "`" + `some_expr{app="bar"}` +"`" + `
lorem ipsum
`
```

And this is already looking confusing and hard to read, therefore I decided to go on with double quotes and escape the one that are in the string. For consistency, the all other string values are using double quotes.

Comment (rfranzke) [reply pkg/component/observability/monitoring/prometheus/garden/prometheusrules.go:91]: I see, fair enough.

**Review by rfranzke — COMMENTED** [review]
/lgtm
